shells/pipes-shell/pipe.js

```diff
@@ -49,63 +48,48 @@ const fromJson = json => {
 };
 
 const extractType = json => {
-  let type = 'com.music.spotify';
-  if (json) {
-    try {
-      const entity = JSON.parse(json) || Object;
-      type = entity.type;
-    } catch (x) {
-      //
-    }
-  }
-  return type;
+  const entity = fromJson(json);
+  return (entity ? entity.type : 'com.music.spotify').replace(/\./g, '_');
 };
 
-const dispatch = (type, arc, callback) => {
+const dispatch = async (type, arc, callback) => {
+  await instantiatePipeRecipe(arc, type);
   switch (type) {
-    case 'com.google.android.apps.maps':
-      com_google_android_apps_maps(arc, callback);
+    case 'com_google_android_apps_maps':
+      await com_google_android_apps_maps(arc, callback);
       break;
-    case 'com.music.spotify':
     default:
-      com_music_spotify(arc, callback);
+    case 'com_music_spotify':
+      await com_music_spotify(arc, callback);
       break;
   }
 };
 
 const com_music_spotify = async (arc, callback) => {
-  await (async () => {
-    const manifestContent = buildEntityManifest({type: 'com_music_spotify'});
-    const manifest = await Utils.parse(manifestContent);
-    await instantiateRecipe(arc, manifest, 'Pipe');
-  })();
-  await (async () => {
-    const manifest = await Utils.parse(`import 'https://$particles/PipeApps/ArtistAutofill.recipes'`);
-    // accrete recipe
-    await instantiateRecipe(arc, manifest, 'ArtistAutofill');
-    // wait for data to appear
-    const store = arc._stores[2];
-    watchOneChange(store, callback, arc);
-    //await dumpStores(arc._stores);
-  })();
+  const manifest = await Utils.parse(`import 'https://$particles/PipeApps/ArtistAutofill.recipes'`);
+  // accrete recipe
+  await instantiateRecipe(arc, manifest, 'ArtistAutofill');
+  // wait for data to appear
+  const store = arc._stores[2];
+  watchOneChange(store, callback, arc);
+  //await dumpStores(arc._stores);
```

Comment:
Delete?

Reply:
The commented code in pipes-shell is there for troubleshooting. Since I can't actually test the integration myself, I don't want to tighten all the screws until we have some confidence.

WRT the commented pipes code in web-shell, I wanted to disable it right away, but will remove it more judiciously after we have some time to discuss the details.